src/id-generation.ts

```diff
@@ -0,0 +1,61 @@
+import {nanoid} from "npm:nanoid@4.0.0"
+import {
+    createHash
+} from "https://deno.land/std@0.168.0/node/internal/crypto/hash.ts"
+
+
+export type IdGenerator = (data?: string) => string
+
+
+export const nanoIdGen = (): IdGenerator =>
+    () => nanoid()
+
+
+export const sequentialIdGen = (): IdGenerator => {
+    let num = 0
+    return () => `${++num}`
+}
+
+
+const defaultHashAlgorithm = "SHA256"
+
+export type IdGenOptions = {
+    doNotCheckForUniqueData?: boolean
```

Comment:
Perhaps we could add comments to indicate which default values will be used, if the properties are not specified

Reply:
I have a branch/PR lined up in which I do exactly that.